client/src/components/LoginSession/LoginSessionContainer.js

```diff
@@ -35,23 +35,24 @@ function LoginSessionContainer(props) {
             SecurityID: Config.get('SecurityID')
         })
         .then(response => {
-            const failure = !response.success;
-            setFailed(failure);
-
-            if (failure) {
-                props.displayToastFailure(response.message);
-            } else {
-                props.displayToastSuccess(response.message);
-            }
+            props.displayToastSuccess(response.message);
         })
-        .catch(() => {
+        .catch((err) => {
             setFailed(true);
-            props.displayToastFailure(i18n._t(
-                'SessionManager.COULD_NOT_LOGOUT',
-                'Could not log out of session. Try again later.'
-            ));
-        })
-        .finally(() => {
+            return err.response.text().then(json => {
+                const response = JSON.parse(json);
+                if (typeof response !== 'object' || typeof response.message !== 'string') {
+                    return Promise.reject('No readable error message');
+                }
+                props.displayToastFailure(response.message);
+                return Promise.resolve();
+            }).catch(() => {
+                props.displayToastFailure(i18n._t(
+                    'SessionManager.COULD_NOT_LOGOUT',
+                    'Could not log out of session. Try again later.'
+                ));
+            });
+        }).finally(() => {
```

Comment:
I refactored this to use proper error handling. Catch statement will handle any scenario where an HTTP Error code is returned.